src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -546,18 +548,63 @@ void Instance::OnFinished(Status status, CharSpan debugText, ThreadScanResponseI
         err = writer->StartContainer(TLV::ContextTag(to_underlying(Commands::ScanNetworksResponse::Fields::kThreadScanResults)),
                                      TLV::TLVType::kTLVType_Array, listContainerType));
 
-    for (; networks != nullptr && networks->Next(scanResponse) && networksEncoded < kMaxNetworksInScanResponse; networksEncoded++)
+    VerifyOrExit(
+        scanResponseArray.Alloc(networks->Count() > kMaxNetworksInScanResponse ? kMaxNetworksInScanResponse : networks->Count()),
+        err = CHIP_ERROR_NO_MEMORY);
+    for (; networks != nullptr && networks->Next(scanResponse);)
+    {
+        if ((scanResponseArrayLength == kMaxNetworksInScanResponse) &&
+            (scanResponseArray[scanResponseArrayLength - 1].rssi > scanResponse.rssi))
+        {
+            continue;
+        }
+
+        bool isDuplicated = false;
+
+        for (size_t i = 0; i < scanResponseArrayLength; i++)
+        {
+            if ((scanResponseArray[i].panId == scanResponse.panId) &&
+                (scanResponseArray[i].extendedPanId == scanResponse.extendedPanId))
+            {
+                isDuplicated = true;
+                if (scanResponseArray[i].rssi < scanResponse.rssi)
+                {
+                    scanResponseArray[i] = scanResponse;
+                    Sorting::InsertionSort(
+                        scanResponseArray.Get(), scanResponseArrayLength,
```

Comment:
```suggestion
                        scanResponseArray.Get(), i + 1,
```
This should be enough, right?

Reply:
Could you please check my other comment?